File: backEnd/src/main/java/com/quiz/ourclass/domain/challenge/dto/response/ChallengeSimpleResponse.java
package com.quiz.ourclass.domain.challenge.dto.response;

import io.swagger.v3.oas.annotations.media.Schema;
import java.time.LocalDateTime;

@Schema(description = "함께달리기 요약 조회 응답 DTO")
public record ChallengeSimpleResponse(
    @Schema(description = "함께달리기 ID")
    long id,
    @Schema(description = "함께달리기 제목")
    String title,
    @Schema(description = "함께달리기 내용")
    String content,
    @Schema(description = "함께달리기 보상")
    int reward,
    @Schema(description = "함께달리기 시작 시간")
    LocalDateTime startTime,
    @Schema(description = "함께달리기 종료 시간")
    LocalDateTime endTime
) {

}

File: android/data/src/main/java/com/sixkids/data/model/response/ChallengeSimpleResponse.kt
package com.sixkids.data.model.response

import com.sixkids.model.Challenge
import java.time.LocalDateTime

data class ChallengeSimpleResponse(
    val id: Int,
    val title: String,
    val content: String,
    val startTime: LocalDateTime,
    val endTime: LocalDateTime,
    val reward: Int,
)

fun ChallengeSimpleResponse.toModel(): Challenge {
    return Challenge(
        id = id,
        title = title,
        content = content,
        startTime = startTime,
        endTime = endTime,
        reword = reward
    )
}
